File: Programmers/codingTest_Kit/Hash/42576.cpp
#include <string>
#include <vector>
#include <unordered_map>
#include <iostream>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> d;
    for(auto& i : participant) d[i]++;
    for(auto& i : completion) d[i]--;
    for(auto& i : d){
        if(i.second > 0){
            answer = i.first;
            break;
        }
    }
    return answer;
}

int main(){
    cout << solution({"leo", "kiki", "eden"}, {"eden", "kiki"}) << '\n';
}

File: Programmers/codingTest_Kit/Hash/42576.py
def solution(participant, completion):
    dic = dict()
    for i in participant:
        if i in dic:
            dic[i] += 1
        else:
            dic[i] = 1
    for i in completion:
        if dic[i] == 1:
            del dic[i]
        else:
            dic[i] -= 1
    return list(dic.keys())[0]